[Enhancement][QoL] Add option to adjust shop overlay opacity

torranx opened the pull request:
## What are the changes?
- see title
- option is only up to 90%

## Why am I doing these changes?
- some players are still not happy with how dark/light the current opacity is
- adding option instead would allow users to customize their experience

## What did change?
- new Shop Overlay Opacity setting

### Screenshots/Videos
- Menu
<img width="1306" alt="Screenshot 2024-07-11 at 11 52 25 PM" src="https://github.com/pagefaultgames/pokerogue/assets/68144167/ca0d53d7-ad7f-4046-8138-7da20a5f6b26">

- 90%
<img width="1300" alt="Screenshot 2024-07-11 at 11 48 44 PM" src="https://github.com/pagefaultgames/pokerogue/assets/68144167/b7db0111-49a4-45d5-a938-7d6dfdcf0972">

- 80% (default)
<img width="1303" alt="Screenshot 2024-07-11 at 11 46 35 PM" src="https://github.com/pagefaultgames/pokerogue/assets/68144167/c145cbb4-2624-4427-a70c-db508c94b9f7">

- 70%
<img width="1302" alt="Screenshot 2024-07-11 at 11 49 10 PM" src="https://github.com/pagefaultgames/pokerogue/assets/68144167/da5ce029-724b-4b9c-bd78-8a445480222a">

- 60%
<img width="1303" alt="Screenshot 2024-07-11 at 11 49 42 PM" src="https://github.com/pagefaultgames/pokerogue/assets/68144167/409ba169-e064-4efd-8154-d5f3b0ec1baa">

- 50%
<img width="1301" alt="Screenshot 2024-07-11 at 11 50 07 PM" src="https://github.com/pagefaultgames/pokerogue/assets/68144167/e3e4681b-0724-483f-b273-867449af4d59">

- 40%
<img width="1298" alt="Screenshot 2024-07-11 at 11 50 36 PM" src="https://github.com/pagefaultgames/pokerogue/assets/68144167/e7142e8d-b384-4a03-a72d-1db282a97100">

- 30%
<img width="1300" alt="Screenshot 2024-07-11 at 11 50 55 PM" src="https://github.com/pagefaultgames/pokerogue/assets/68144167/de5be715-6937-41da-ab41-c701914067d5">

- 20%
<img width="1300" alt="Screenshot 2024-07-11 at 11 51 19 PM" src="https://github.com/pagefaultgames/pokerogue/assets/68144167/abe15de0-4146-452f-96c1-feddca986ff4">

- 10%
<img width="1303" alt="Screenshot 2024-07-11 at 11 51 39 PM" src="https://github.com/pagefaultgames/pokerogue/assets/68144167/9da4381a-449b-435a-a083-e8efce7ece05">


## How to test the changes?
- get to shop phase
- change overlay value and check the change

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [ ] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

Adri1 commented on src/locales/fr/settings.ts:
```diff
@@ -96,4 +96,6 @@ export const settings: SimpleTranslationEntries = {
   "controller": "Controller",
   "gamepadSupport": "Gamepad Support",
   "showBgmBar": "Montrer titre de la musique",
+  "none": "None",
+  "shopOverlayOpacity": "Shop Overlay Opacity"
```
```suggestion
  "none": "Aucune",
  "shopOverlayOpacity": "Opacité boutique"
```

Adri1 reviewed (commented):
Added French

40chyan commented on src/locales/zh_CN/settings.ts:
```diff
@@ -96,4 +96,6 @@ export const settings: SimpleTranslationEntries = {
   "controller": "控制器",
   "gamepadSupport": "手柄支持",
   "showBgmBar": "显示音乐名称",
+  "none": "None",
+  "shopOverlayOpacity": "Shop Overlay Opacity"
```
```suggestion
  "none": "无",
  "shopOverlayOpacity": "商店显示不透明度"
```

Adri1 commented on src/locales/fr/settings.ts:
```diff
@@ -96,4 +96,6 @@ export const settings: SimpleTranslationEntries = {
   "controller": "Controller",
   "gamepadSupport": "Gamepad Support",
   "showBgmBar": "Montrer titre de la musique",
```
```suggestion
  "showBgmBar": "Titre de la musique",
```

Dakurei reviewed (approved):
Tested locally, no particular errors encountered during testing 👍

torranx commented:
Updated values to be 20 30 40 50 60 70 80 90

- Removed `0` as this would not make sense to have.
- Removed `10` as well as this is visually the same with `0`

josericardo-fo commented on src/locales/pt_BR/settings.ts:
```diff
@@ -96,4 +96,5 @@ export const settings: SimpleTranslationEntries = {
   "controller": "Controle",
   "gamepadSupport": "Suporte para Controle",
   "showBgmBar": "Show Music Names",
+  "shopOverlayOpacity": "Shop Overlay Opacity"
```
```suggestion
  "shopOverlayOpacity": "Opacidade da Loja"
```

EnochG1 commented on src/locales/ko/settings.ts:
```diff
@@ -96,4 +96,5 @@ export const settings: SimpleTranslationEntries = {
   "controller": "컨트롤러",
   "gamepadSupport": "게임패드 지원",
   "showBgmBar": "BGM 제목 보여주기",
+  "shopOverlayOpacity": "Shop Overlay Opacity"
```
```suggestion
  "shopOverlayOpacity": "상점 오버레이 투명도"
```

EnochG1 reviewed (commented):
Korean translation

InnocentGameDev commented on src/locales/es/settings.ts:
```diff
@@ -96,4 +96,5 @@ export const settings: SimpleTranslationEntries = {
   "controller": "Controller",
   "gamepadSupport": "Gamepad Support",
   "showBgmBar": "Show Music Names",
+  "shopOverlayOpacity": "Shop Overlay Opacity"
```
```suggestion
  "shopOverlayOpacity": "Opacidad de la fase de compra"
```

InnocentGameDev reviewed (commented):
Es localization

Tempo-anon reviewed (commented):
Do you mind adding screenshots for how the same shop would look like at each opacity?

torranx commented:
> Do you mind adding screenshots for how the same shop would look like at each opacity?

updated PR description
